stan/math/opencl/kernel_generator/rowwise_reduction.hpp

```diff
@@ -16,6 +18,60 @@
 
 namespace stan {
 namespace math {
+namespace internal {
+
+template <typename Arg>
+struct matvec_mul_opt {
+  enum { is_possible = 0 };
+
+  static matrix_cl_view view(const Arg&) { return matrix_cl_view::Entire; }
+
+  static kernel_parts get_kernel_parts(
+      const Arg& a, std::set<const operation_cl_base*>& generated,
+      name_generator& name_gen, const std::string& i, const std::string& j) {
+    return {};
+  }
+};
+
+template <typename Mat, typename VecT>
+struct matvec_mul_opt<
+    elewise_multiplication_<Mat, broadcast_<VecT, true, false>>> {
+  enum { is_possible = 1 };
+  using Arg = elewise_multiplication_<Mat, broadcast_<VecT, true, false>>;
+
+  static matrix_cl_view view(const Arg& a) {
+    return a.template get_arg<1>().template get_arg<0>().view();
+  }
+
+  static kernel_parts get_kernel_parts(
+      const Arg& mul, std::set<const operation_cl_base*>& generated,
+      name_generator& name_gen, const std::string& i, const std::string& j) {
+    kernel_parts res{};
+    if (generated.count(&mul) == 0) {
+      mul.var_name = name_gen.generate();
+      generated.insert(&mul);
+
+      const auto& matrix = mul.template get_arg<0>();
+      const auto& broadcast = mul.template get_arg<1>();
+      res = matrix.get_kernel_parts(generated, name_gen, i, j, true);
+      if (generated.count(&broadcast) == 0) {
+        broadcast.var_name = name_gen.generate();
+        generated.insert(&broadcast);
+
+        const auto& vec_t = broadcast.template get_arg<0>();
```

Comment:
We reserve `_t` stuff for type traits. Maybe `vec_expr`

Reply:
fixed